SportsGoods.Data/Repositories/ProductRepository.cs

```diff
@@ -43,6 +43,13 @@ public async Task<PagedResult<Product>> GetPagedAsync(int pageNumber, byte pageS
               .Take(pageSizeInt)
               .ToListAsync();
 
+
+
+            foreach (var product in products)
+            {
+                product.Brand = await _context.Brands.FindAsync(product.BrandId);
```

Comment:
this one is going to be slow 
because you are going to make n number of queries in the db (where n is the count of products)
you should obtain all of them in a single query operation 
you already know all brand ids so get rid of the foreach 
@LVarsaminov

Reply:
Done